### PR TITLE
Add cluster state to content bucket space.

### PR DESCRIPTION
**Pull request by toregge**
@geirst, @vekterli: please review

### CODE REVIEW

File: storage/src/vespa/storage/common/content_bucket_space.cpp [hunk 1]
```diff
@@ -8,10 +8,25 @@ ContentBucketSpace::ContentBucketSpace(document::BucketSpace bucketSpace)
...
       _lock(),
+      _clusterState(),
       _distribution()
 {
 }
 
+void
+ContentBucketSpace::setClusterState(std::shared_ptr<const lib::ClusterState> clusterState)
+{
+    std::lock_guard<std::mutex> guard(_lock);
```
Comment (vekterli) [inline storage/src/vespa/storage/common/content_bucket_space.cpp]: Can just write `std::lock_guard` without explicit type in C++17

File: storage/src/vespa/storage/common/content_bucket_space.cpp [hunk 2]
```diff
@@ -8,10 +8,25 @@ ContentBucketSpace::ContentBucketSpace(document::BucketSpace bucketSpace)
...
+    _clusterState = std::move(clusterState);
+}
+
+std::shared_ptr<const lib::ClusterState>
+ContentBucketSpace::getClusterState() const
+{
+    std::lock_guard<std::mutex> guard(_lock);
```
Comment (vekterli) [inline storage/src/vespa/storage/common/content_bucket_space.cpp]: Also here

**Review by vekterli — APPROVED** [review]
:+1: only type signature simplification nitpicks

**Review by geirst — APPROVED** [review]
👍

**Pull request merged**